### PR TITLE
Add initial implementation

### PR DESCRIPTION
**Pull request by klzns**

### CODE REVIEW

File: example/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+```js
+"product-summary.shelf": {
+  "children": [
+    // ...other blocks
+    "product-rating-inline",
+    // ...other blocks
+  ]
+}
+```
```
Comment (victorhmp) [inline example/README.md]: Shouldn't this be describing a Wishlist app?

File: manifest.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "vendor": "vtex",
+  "name": "product-bookmark-interfaces",
+  "version": "1.0.0",
```
Comment (victorhmp) [inline manifest.json:4]: Why not `0.x` major?

File: react/Noop.tsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import React, { FunctionComponent } from 'react'
+
+const Noop: FunctionComponent = () => <></>
```
Comment (victorhmp) [inline react/Noop.tsx:3]: Would logging an error to the console be useful? I'm thinking of something similar to what is done in the `flex-layout` app: https://github.com/vtex-apps/flex-layout/blob/master/react/FlexLayoutDummy.js

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/.prettierrc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (kaisermann) [inline react/.prettierrc]: Switch the whole content for `"@vtex/prettier-config"` 💅

File: react/package.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "@types/graphql": "^14.2.0",
+    "@types/jest": "^24.0.11",
+    "@types/node": "^11.13.0",
+    "@types/prop-types": "^15.7.0",
+    "@types/react": "^16.8.10",
+    "apollo-client": "^2.5.1",
+    "eslint": "^5.16.0",
+    "eslint-config-vtex": "^10.1.0",
+    "prettier": "^1.16.4",
+    "tslint-eslint-rules": "^5.4.0",
```
Comment (kaisermann) [inline react/package.json]: Is `tslint-eslint-rules` used?

File: example/.prettierrc [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (kaisermann) [inline example/.prettierrc]: I know that's an example, but assuming people will copy those file to create their projects: replace all the content with `"@vtex/prettier-config"` 🙏

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Added
+
+- Initial release
```
Comment (kaisermann) [inline CHANGELOG.md]:
```suggestion
- Initial release.
```

File: example/react/ProductBookmark.tsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  const { formatMessage } = useIntl()
+  const [filled, setFill] = useState(false)
+  const productContext = useProduct()
+
+  const handleClick = (ev: MouseEvent) => {
+    ev.stopPropagation()
+    ev.preventDefault()
+    setFill(!filled)
+    if (productContext && productContext.product) {
+      // eslint-disable-next-line no-console
```
Comment (kaisermann) [inline example/react/ProductBookmark.tsx:32]: Is this `console.log` supposed to be here?

**Review by kaisermann — CHANGES_REQUESTED** [review]

File: react/Noop.tsx (hunk 3 above)
Comment (klzns) [reply react/Noop.tsx:3]: Nope. Check the timeline:

1.  The theme would add those empty blocks
2. When opening the store, no consoles should appear
3. Install the wishlist app
4. Store builder would plug the blocks in the blocks of step 1
5. When opening the store, the blocks of the wishlist are in place

File: example/react/ProductBookmark.tsx (hunk 8 above)
Comment (klzns) [reply example/react/ProductBookmark.tsx:32]: yess, as an example of how to use data from context

File: manifest.json (hunk 2 above)
Comment (klzns) [reply manifest.json:4]: Because it's public

File: react/Noop.tsx (hunk 3 above)
Comment (victorhmp) [reply react/Noop.tsx:3]: Oooh! I see! Awesome then 😄

**Review by victorhmp — APPROVED** [review]

**Review by kaisermann — APPROVED** [review]

**Pull request merged**